test/widgets/login_test.dart

```diff
@@ -61,7 +64,44 @@ void main() {
     expectErrorFromText('email@example.com', ServerUrlValidationError.noUseEmail);
   });
 
-  // TODO test AddAccountPage
+  group('AddAccountPage server URL helper text', () {
+    Future<void> prepareAddAccountPage(WidgetTester tester) async {
+      await tester.pumpWidget(const MaterialApp(
+        localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+        supportedLocales: ZulipLocalizations.supportedLocales,
+        home: AddAccountPage(),
+      ));
+    }
+
+    final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+
+    Future<Finder> findHelperText(WidgetTester tester) async {
+      return find.text(zulipLocalizations.serverUrlDocLinkLabel);
+    }
+
+    testWidgets('launches URL when helper text is tapped', (tester) async {
+      await prepareAddAccountPage(tester);
+      await tester.tap(await findHelperText(tester));
+
+      check(testBinding.takeLaunchUrlCalls()).single.equals((
+        url: AddAccountPage.serverUrlHelpUrl,
+        mode: LaunchMode.platformDefault,
+      ));
+    });
+
+    testWidgets('shows error dialog when URL fails to open', (tester) async {
+      await prepareAddAccountPage(tester);
+      testBinding.launchUrlResult = false;
```

Comment:
When a test manipulates `testBinding`, it needs to call `addTearDown(testBinding.reset)` somewhere so that the state gets reset before the next test.

(In the "error dialog if invalid link" in widgets/content_test.dart which this one is presumably based on, that gets taken care of by the `prepare` helper, because the latter does other things with `testBinding` that also need a reset.)

You can reproduce the problem with the command:
```
$ flutter test test/widgets/login_test.dart --test-randomize-ordering-seed=random
```
At your current revision, that often fails because `launchUrlResult` remains set to false when the other new test tries to run. After you fix the problem with a reset, that command should pass every time.